bench.js

```diff
@@ -8,20 +8,16 @@ var globby = require('./');
 
 var BENCH_DIR = 'bench';
 
-function globbyToCb(globbyFn) {
-	return function (patterns, cb) {
-		setImmediate(function () {
-			globbyFn(patterns).then(cb.bind(null, null), cb);
-		});
-	};
-}
-
 var runners = [{
 	name: 'globby async (working directory)',
-	run: globbyToCb(globby)
+	run: function (patterns, cb) {
+		globby(patterns).then(cb.bind(), cb);
```

Comment:
I think you still need `.bind(null, null)` (bind `null` to the callback's `this` and `err` argument), otherwise the promise's fulfillment value will be passed as the error argument to matcha's callback, isn't it?